reporters/dogstatsd.go

```diff
@@ -84,3 +109,21 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) restartTicker() {
+	for range d.ticker.C {
+		d.mutex.Lock()
+		if err := d.statsdClient.Close(); err != nil {
```

Comment:
1. Why does this statsdClient exist if it isn't being used besides this close here?
2. Why is the lock/unlock here? We need to lock/unlock the client being used to report, this one exists just to be closed?

Reply:
1. Because the current implementation of maestro wraps the interface from Datadog's client implemented in the TFGCo Go Extensions lib, which in turn also hides the original implementation of `statsd.Client`. So, to be able to access the `Close` function on the restartTicker, I need to cast in the constructor of maestro-Datadog's client ([here](https://github.com/topfreegames/maestro/blob/v9/reporters/dogstatsd.go#L73)). 
    You can see the wrapper of Datadog's client from TFGCo Go Extensions [here](https://github.com/topfreegames/extensions/blob/93d8d3815c98908f0fbb0e3050fdadd135bf9ede/dogstatsd/dogstatsd.go#L17).
2. Ops! Actually, we should lock the report functions too! I'll fix it soon!

Reply:
Added locking in report function.

Reply:
I think this mutex should include this whole function, right? Even the part that assigns the new client. Otherwise, the report function could use a closed reporter, leading to an error. 

What I suggest is:
```suggestion
                defer d.mutex.Unlock()
		if err := d.statsdClient.Close(); err != nil {
```

And remove the Unlock below. So, it'll Unlock at every return of the function.

Reply:
Deferring, in this case, will unlock the Mutex only when the function call is finished. Since we're using a for loop over a ticker, does the `defer` close after the iteration over a tick?

Reply:
That works. Don't worry about changing it, but another idea was to separate in another method. So you could do:
```
for {
  mutex.Lock()
  submethod()
  mutex.Unlock()  
}
```
But that is more useful for methods with many returns since the developer doesn't need to add the unlock in all of them, with the risk of forgetting one.